code/modules/explorer_drone/adventure.dm

```diff
@@ -41,25 +42,133 @@
+	var/id
+	/// actual adventure json string
+	var/raw_json
+	/// ckey of last change user.
+	var/uploader
+	/// Time of last change.
+	var/timestamp
+	/// Unapproved adventures won't be used for exploration sites.
+	var/approved = FALSE
+	/// Was the adventure used for exploration site this round.
+	var/placed = FALSE
+
+	//Variables below are extracted from the JSON
 
-/proc/try_loading_adventure(filename)
-	var/list/json_data = json_load(ADVENTURE_DIR+filename)
+	/// json version
+	var/version
+	/// adventure name
+	var/name
+	/// required site traits to use this adventure
+	var/list/required_site_traits
+
+/// Check if the adventure usable for given exploration site traits
+/datum/adventure_db_entry/proc/valid_for_use(list/site_traits)
+	if(!raw_json || version != CURRENT_ADVENTURE_VERSION || placed)
+		return FALSE
+	if(required_site_traits && length(required_site_traits - site_traits) != 0)
+		return FALSE
+	return TRUE
+
+/// Creates DB entry if it does not exist in the DB yet. Otherwise updates local data if our timestamp is older or updates db data if our timestamp is newer.
+/datum/adventure_db_entry/proc/synchronize()
+	if(id)
+		//Check if our timestamp is fresh, if not update local and stop
+		var/datum/db_query/TimestampCheckQuery = SSdbcore.NewQuery("SELECT timestamp > :current_timestamp,adventure_data,uploader,timestamp,approved FROM [format_table_name("text_adventures")] WHERE id = :id",list("id" = id, "current_timestamp" = timestamp))
+		if(!TimestampCheckQuery.warn_execute() || !TimestampCheckQuery.NextRow())
+			qdel(TimestampCheckQuery)
+			return
+		if(!timestamp || text2num(TimestampCheckQuery.item[1]) > 0)
+			raw_json = TimestampCheckQuery.item[2]
+			uploader = TimestampCheckQuery.item[3]
+			timestamp = TimestampCheckQuery.item[4]
+			approved = TimestampCheckQuery.item[5]
+			extract_metadata()
+			qdel(TimestampCheckQuery)
+			return
+		qdel(TimestampCheckQuery)
+		//We're up to date, update db instead
+		var/datum/db_query/UpdateQuery = SSdbcore.NewQuery("UPDATE [format_table_name("text_adventures")] SET adventure_data = :adventure_data,uploader = :uploader,approved = :approved WHERE id = :id",
+		list("id" = id, "adventure_data" = raw_json, uploader = usr.ckey, approved = approved))
+		UpdateQuery.warn_execute()
+		qdel(UpdateQuery)
+	else
+		// Create new entry
+		var/datum/db_query/InsertQuery = SSdbcore.NewQuery("INSERT INTO [format_table_name("text_adventures")] (adventure_data, uploader) VALUES (:raw_json, :uploader)", list("raw_json" = raw_json, "uploader" = usr.ckey))
+		if(!InsertQuery.warn_execute())
+			qdel(InsertQuery)
+			return FALSE
+		uploader = usr.ckey
+		id = InsertQuery.last_insert_id
+		qdel(InsertQuery)
+
+/// Deletes the local AND db entry.
+/datum/adventure_db_entry/proc/remove()
+	if(id)
+		var/datum/db_query/DelQuery = SSdbcore.NewQuery("DELETE FROM [format_table_name("text_adventures")] WHERE id = :id", list("id" = id))
+		if(DelQuery.warn_execute())
```

Comment:
I think you should invert this and do the usual
```
if(!DelQuery.warn_execute())
	qdel(DelQuery)
	return FALSE
```
as otherwise it's possible to have a situation where the query fails to execute (in this case basically only due to a connection error) and no log message is created but the proc execution continues and still removes the datum from `GLOB.explorer_drone_adventure_db_entries` at L142